src/Authenticator/ChangePasswordHandler.php

```diff
@@ -0,0 +1,190 @@
+            return $this->redirectBack();
+        }
+
+        $this->applyRequirements();
+
+        return [
+            'Form' => $this->renderWith($this->getViewerTemplates()),
+            'ClassName' => 'mfa',
+        ];
+    }
+
+    /**
+     * Initiates the session for the user attempting to log in, in preparation for an MFA check
+     *
+     * @param HTTPRequest $request
+     * @return HTTPResponse
+     */
+    public function startMFACheck(HTTPRequest $request): HTTPResponse
+    {
+        $store = $this->getStore();
+        if (!$store) {
+            throw new LogicException('Store not found, please create one first.');
+        }
+        $member = $store->getMember();
+
+        // If we don't have a valid member we shouldn't be here...
+        if (!$member) {
+            return $this->jsonResponse(['message' => 'Forbidden'], 403);
+        }
+
+        // Use the provided trait method for handling login
+        $response = $this->createStartLoginResponse(
+            $store,
+            Injector::inst()->get(MethodRegistry::class)->getMethodByURLSegment($request->param('Method'))
+        );
+
+        // Ensure detail is saved to the store
+        $store->save($request);
+
+        return $response;
+    }
+
+    /**
+     * Checks the MFA JavaScript app input to validate the user attempting to log in
+     *
+     * @param HTTPRequest $request
+     * @return HTTPResponse
+     */
+    public function verifyMFACheck(HTTPRequest $request): HTTPResponse
+    {
+        $store = $this->getStore();
+
+        try {
+            $result = $this->verifyLoginRequest($store, $request);
+        } catch (InvalidMethodException $exception) {
+            // Invalid method usually means a timeout. A user might be trying to verify before "starting"
+            Injector::inst()->get(LoggerInterface::class . '.mfa')->info($exception->getMessage());
+            return $this->jsonResponse(['message' => 'Forbidden'], 403);
+        }
+
+        if (!$result->isSuccessful()) {
+            return $this->jsonResponse([
+                'message' => $result->getMessage(),
+            ], 401);
+        }
+
+        if (!$this->isLoginComplete($store)) {
+            return $this->jsonResponse([
+                'message' => 'Additional authentication required',
+            ], 202);
+        }
+
+        $this->getRequest()->getSession()->set(self::MULTIFACTORAUTHENTICATED, true);
+        $store->clear($request);
+
+        return $this->jsonResponse([
+            'message' => 'Multi factor authenticated',
+        ], 200);
+    }
+
+    public function changepassword()
+    {
+        $session = $this->getRequest()->getSession();
+        $hash = $session->get('AutoLoginHash');
+        $member = Member::member_from_autologinhash($hash);
+        if ($hash
+            && $member
+            && $member->RegisteredMFAMethods()->exists()
+            && !$session->get(self::MULTIFACTORAUTHENTICATED)) {
```

Comment:
```suggestion
            && !$session->get(self::MULTIFACTORAUTHENTICATED)
        ) {
```

I'm surprised this doesn't fail in linting, it should